### PR TITLE
upgrade dps-gradle-spring-boot to 4.1.1

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

upgrade dps-gradle-spring-boot to 4.1.1

## What is the intent behind these changes?

pull bundled `.trivyignore` to fix trivy scanning failures.

### CODE REVIEW

**Review by rgforsyth — APPROVED** [review]

**Comment by rgforsyth** [comment]
Approved change but seems to be failing vulnerability scan

File: .trivyignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ #           AND ANY MANUAL CHANGES WILL BE OVERRIDDEN ON YOUR NEXT BUILD.
+ #
+ # To make general changes to the suppressions below, change the gradle plugin dps-gradle-spring-boot,
+ # publish a new version and update to the new version in your gradle build script
+ #
+ # To stop the dps-gradle-spring-boot project from overwriting any project specific customisations here, remove the
+ # warning at the top of this file.
+
+ # suppress jackson warning until 2.14 is published
+ CVE-2020-36518
```
Comment (sldblog) [inline .trivyignore:11]: I don't think this will pacify `hmpps/gradle_owasp_dependency_check` 💭

Comment (sldblog) [reply .trivyignore:11]: Oh and also that the nightly scans work on the latest (dev) image, rather than the one in production ☘️

**Comment by sldblog** [comment]
Shall we increase trivy threshold to MEDIUM and disable `hmpps/gradle_owasp_dependency_check`?
@ministryofjustice/hmpps-interventions-dev

**Pull request merged**